geniza/corpus/tests/test_corpus_templates.py

```diff
@@ -656,3 +658,42 @@ def test_related_list(self, client, document, join, empty_solr):
 
         # "join" fixture should be in list
         assertContains(response, f"<dd>{join.id}</dd>")
+
+
+class TestFieldsetSnippet:
+    """Unit tests for the override of django admin/includes/fieldset.html, which allows
+    inclusion of inline formsets between model form fields"""
+
+    template = "admin/corpus/document/snippets/fieldset.html"
+
+    def test_inlines_included(self, admin_client, document):
+        # the snippet should be included on the admin document change page
+        response = admin_client.get(
+            reverse("admin:corpus_document_change", args=(document.id,))
+        )
+        assertTemplateUsed(response, template_name=self.template)
+
+        # should include Dating inline
+        assertTemplateUsed(response, template_name=DocumentDatingInline.template)
+        assertContains(
+            response,
+            'div class="js-inline-admin-formset inline-group" id="dating_set-group"',
+        )
+
+        # Dating inline should be immediately after standard_date field (which is the value of
+        # DocumentDatingInline.insert_after)
+        soup = BeautifulSoup(response.content)
+        standard_date_field = soup.find(
+            "div", class_=f"fieldBox field-{DocumentDatingInline.insert_after}"
+        )
+        assert standard_date_field.find_next_sibling("div")["id"] == "dating_set-group"
+        dating_inline = soup.find("div", id="dating_set-group")
+        assert dating_inline.find_parent("fieldset") is not None
```

Comment:
whoa, hadn't thought about checking sequence like this; I suspect if it breaks it will not be present at all, but this seems like a useful way to bring this to our attention in future upgrades if something changes!